### PR TITLE
Stop the stability checker running on multiple commits on master.

### PR DESCRIPTION
**Pull request by jgraham**
The stability checker tried to work out which commits to test by
looking at all the commits on the current branch that aren't on
another branch. But if the current branch is e.g. master then this
will produce many more commits than intended. To fix this use the
merge-base with master when it's closer to HEAD than the last commit
not on another branch.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5196/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/213807706)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 87edb59c7438ebe3b8796d634b1ae0e74ef1d15e
Using browser at version BuildID 20170321110237; SourceStamp ca4ae502156eaea6fffb296bb9c3b3930af8ab58
Using browser at version BuildID 20170321110237; SourceStamp ca4ae502156eaea6fffb296bb9c3b3930af8ab58
Starting 10 test iterations
Starting 10 test iterations
No tests run.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/213807707)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 87edb59c7438ebe3b8796d634b1ae0e74ef1d15e
Using browser at version 59.0.3047.0 dev
Using browser at version 59.0.3047.0 dev
Starting 10 test iterations
Starting 10 test iterations
No tests run.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/213904513)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 13094b03638ea1cd029c4ac8e58506a0df556284
Using browser at version BuildID 20170321110237; SourceStamp ca4ae502156eaea6fffb296bb9c3b3930af8ab58
Using browser at version BuildID 20170321110237; SourceStamp ca4ae502156eaea6fffb296bb9c3b3930af8ab58
Starting 10 test iterations
Starting 10 test iterations
No tests run.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/213904515)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 13094b03638ea1cd029c4ac8e58506a0df556284
Using browser at version 59.0.3047.0 dev
Using browser at version 59.0.3047.0 dev
Starting 10 test iterations
Starting 10 test iterations
No tests run.

File: check_stability.py [hunk 1]
```diff
@@ -408,6 +408,15 @@ def get_branch_point(user):
         commits = git("rev-list", "HEAD", *not_heads).split("\n")
         first_commit = commits[-1]
         branch_point = git("rev-parse", first_commit + "^")
+        # The above can produce a too-early commit if we are e.g. on master and there are
+        # preceeding changes that were rebased and so aren't on any other branch. To avoid
```
Comment (jugglinmike) [inline check_stability.py]: Typo: "preceding"

**Review by jugglinmike — CHANGES_REQUESTED** [review]
What a complicated situation! I verified this locally by:

1. applying your changes to my local `master` branch
2. disabling the `fetch_wpt` function
3. adding two empty commits to my local `master` branch
4. pushing my local `master` branch to my "origin" remote (my fork of the project)
5. running the stability checker

In this case, the stability checker identified the "branch point" as the first of
the two commits I added in step 3 (whereas with the change disabled, it
identifies the *parent* of that commit).

There's just one small technical issue to resolve.

**Comment by jugglinmike** [comment]
Thanks, James!

**Pull request merged**